x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/action/TransportPutDataFrameTransformAction.java

```diff
@@ -213,15 +215,19 @@ private void putDataFrame(Request request, ActionListener<AcknowledgedResponse>
         ActionListener<Boolean> pivotValidationListener = ActionListener.wrap(
             validationResult -> dataFrameTransformsConfigManager.putTransformConfiguration(config, putTransformConfigurationListener),
             validationException -> listener.onFailure(
-                    new RuntimeException(DataFrameMessages.REST_PUT_DATA_FRAME_FAILED_TO_VALIDATE_DATA_FRAME_CONFIGURATION,
+                    new ElasticsearchStatusException(
+                        DataFrameMessages.REST_PUT_DATA_FRAME_FAILED_TO_VALIDATE_DATA_FRAME_CONFIGURATION,
+                        RestStatus.BAD_REQUEST,
                         validationException))
         );
 
         try {
             pivot.validateConfig();
-        } catch (Exception e) {
+        } catch (ElasticsearchException e) {
             listener.onFailure(
-                new RuntimeException(DataFrameMessages.REST_PUT_DATA_FRAME_FAILED_TO_VALIDATE_DATA_FRAME_CONFIGURATION,
+                new ElasticsearchStatusException(
+                    DataFrameMessages.REST_PUT_DATA_FRAME_FAILED_TO_VALIDATE_DATA_FRAME_CONFIGURATION,
+                    RestStatus.BAD_REQUEST,
```

Comment:
this overwrites whatever is defined by the catched exception. It think we should forward the rest status of the catched exception.

Reply:
This assumes that the underlying exception throws a rest status. They were previously runtime exceptions, which did nothing of the sort. I can adjust so that the underlying query validation throws status exceptions.

Reply:
`ElasticSearchException` has `status()` so you have a status you can use. Let me elaborate the suggestion: 

```
try {
    pivot.validateConfig();
} catch (ElasticsearchException e) {
// expected errors, e.g. user error
    listener.onFailure(new ElasticsearchStatusException(
      DataFrameMessages.REST_PUT_DATA_FRAME_FAILED_TO_VALIDATE_DATA_FRAME_CONFIGURATION,
      e.status(), e));
} catch (Exception e) {
// for the unexpected error, e.g. bugs (feel free to use a different message text)
    listener.onFailure(new ElasticsearchStatusException(
    DataFrameMessages.REST_PUT_DATA_FRAME_FAILED_TO_VALIDATE_DATA_FRAME_CONFIGURATION,
    RestStatus.INTERNAL_SERVER_ERROR,
    e));
}

Reply:
Yes, I did something similar in the latest commit.